diff --git a/sites/all/modules/custom/devversions/views/js/ajax_view.js b/sites/all/modules/custom/devversions/views/js/ajax_view.js
index ce6cc52..ded94bb 100644
--- a/sites/all/modules/custom/devversions/views/js/ajax_view.js
+++ b/sites/all/modules/custom/devversions/views/js/ajax_view.js
@@ -46,7 +46,8 @@ Drupal.views.ajaxView = function(settings) {
   }
 
   this.element_settings = {
-    url: ajax_path + queryString,
+    //url: ajax_path + queryString,
+    url: ajax_path,
     submit: settings,
     setClick: true,
     event: 'click',
